drivers/SmartThings/matter-switch/src/init.lua

```diff
@@ -144,92 +221,200 @@ local function assign_child_profile(device, child_ep)
+        device.log.debug("configuring for long press device")
+      elseif tbl_contains(MSR, ep) then
+        device.log.debug("configuring for emulated held")
+        set_field_for_endpoint(device, EMULATE_HELD, ep, true, {persist = true})
+      else -- device only supports momentary switch, no release events
+        device.log.debug("configuring for press event only")
+        supportedButtonValues_event = capabilities.button.supportedButtonValues({"pushed"}, {visibility = {displayed = false}})
+        set_field_for_endpoint(device, INITIAL_PRESS_ONLY, ep, true, {persist = true})
+      end
+
+      if supportedButtonValues_event then
+        device:emit_event_for_endpoint(ep, supportedButtonValues_event)
+      end
+      device:emit_event_for_endpoint(ep, capabilities.button.button.pushed({state_change = false}))
+    end
+  end
+end
+
+local function find_child(parent, ep_id)
+  return parent:get_child_by_parent_assigned_key(string.format("%d", ep_id))
+end
+
 local function initialize_switch(driver, device)
   local switch_eps = device:get_endpoints(clusters.OnOff.ID)
-  table.sort(switch_eps)
-  -- Since we do not support bindings at the moment, we only want to count On/Off
-  -- clusters that have been implemented as server. This can be removed when we have
-  -- support for bindings.
-  local num_server_eps = 0
-  local main_endpoint = find_default_endpoint(device)
-  for _, ep in ipairs(switch_eps) do
-    if device:supports_server_cluster(clusters.OnOff.ID, ep) then
-      num_server_eps = num_server_eps + 1
-      if ep ~= main_endpoint then -- don't create a child device that maps to the main endpoint
-        local name = string.format("%s %d", device.label, num_server_eps)
-        local child_profile = assign_child_profile(device, ep)
-        driver:try_create_device(
-          {
-            type = "EDGE_CHILD",
-            label = name,
-            profile = child_profile,
-            parent_device_id = device.id,
-            parent_assigned_child_key = string.format("%d", ep),
-            vendor_provided_label = name
-          }
-        )
+  local button_eps = device:get_endpoints(clusters.Switch.ID, {feature_bitmap=clusters.Switch.types.Feature.MOMENTARY_SWITCH})
+  local all_eps = {}
+
+  local new_profile = nil
+
+  local component_map = {}
+  local current_component_number = 2
+  local component_map_used = false
+  local parent_child_device = false
+
+  if #switch_eps > 0 or #button_eps > 0 then
+    for _,v in ipairs(switch_eps) do
+      table.insert(all_eps, v)
+    end
+    for _,v in ipairs(button_eps) do
+      table.insert(all_eps, v)
+    end
+    table.sort(all_eps)
+    -- Since we do not support bindings at the moment, we only want to count clusters
+    -- that have been implemented as server. This can be removed when we have
+    -- support for bindings.
+    local num_switch_server_eps = 0
+    local main_endpoint = find_default_endpoint(device)
+
+    for _, ep in ipairs(all_eps) do
```

Comment:
Since this change does not add support for combination switch-button device types, I don't know if we should be considering `all_eps` when going through the initialization. I understand that the combination functionality will be added in a future change, but currently this solution seems to provide some support for combination devices by combining eps, which could potentially result in unexpected pathways when onboarding a combination device that may result in matching to a profile that doesn't exist. It may be easier to just separate the switch and button ep handling all together such that if switch_eps > 0, then handle making switch child devices. elseif button eps > 0, then make button endpoint to component maps. In this case, switch takes precendent such that if a device supports both switch and button EPs, only the switch ones would be present. But it seems potentially buggy to combine the ep handling in this init function when we do not support combination devices throughout the rest of the driver.

Reply:
Addressed in 56f40cf534af547dee93afcff6d524d82419be86. I separated the initialization of switch and button eps as suggested. This also helped consolidate things a bit.

Reply:
Let me know if the changes line up with what you had in mind!